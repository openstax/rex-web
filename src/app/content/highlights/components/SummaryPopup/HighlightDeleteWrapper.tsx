import React from 'react';
import { FormattedMessage } from 'react-intl';
import styled from 'styled-components/macro';
import Button from '../../../../components/Button';
import { textStyle } from '../../../../components/Typography';
import { useDrawFocus } from '../../../../reactUtils';
import { HighlightEditButtons } from './styles';

// tslint:disable-next-line:variable-name
const StyledHighlightDeleteWrapper = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(0, 0, 0, 0.9);

  span {
    ${textStyle}
    font-size: 16px;
    font-weight: 500;
    line-height: 25px;
    letter-spacing: -0.2;
    color: #fff;
    margin-right: 8px;
  }
`;

interface HighlightDeleteWrapperProps {
  hasAnnotation?: boolean;
  onCancel: () => void;
  onDelete: () => void;
}

// tslint:disable-next-line:variable-name
const HighlightDeleteWrapper = ({
  hasAnnotation,
  onDelete,
  onCancel,
}: HighlightDeleteWrapperProps) => <StyledHighlightDeleteWrapper
  data-analytics-region='MH delete'
  tabIndex={0}
  ref={useDrawFocus()}
>
  <FormattedMessage
    id={hasAnnotation
      ? 'i18n:highlighting:confirmation:delete-both'
      : 'i18n:highlighting:confirmation:delete-highlight'}
  >
    {(msg) => <span>{msg}</span>}
  </FormattedMessage>
  <HighlightEditButtons>
    <FormattedMessage id='i18n:highlighting:button:delete'>
      {(msg) => <Button
        data-testid='delete'
        data-analytics-label='delete'
        size='medium'
        variant='primary'
        onClick={onDelete}
      >{msg}</Button>}
    </FormattedMessage>
    <FormattedMessage id='i18n:highlighting:button:cancel'>
      {(msg) => <Button
        size='medium'
        data-analytics-label='cancel'
        data-testid='cancel'
        onClick={onCancel}
      >{msg}</Button>}
    </FormattedMessage>
  </HighlightEditButtons>
</StyledHighlightDeleteWrapper>;

export default HighlightDeleteWrapper;
