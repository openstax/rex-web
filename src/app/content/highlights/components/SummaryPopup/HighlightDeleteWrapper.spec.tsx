import React from 'react';
import renderer from 'react-test-renderer';
import TestContainer from '../../../../../test/TestContainer';
import HighlightDeleteWrapper from './HighlightDeleteWrapper';

describe('HighlightDeleteWrapper', () => {
  it('match snapshot', () => {
    const component = renderer.create(<TestContainer>
      {/* tslint:disable-next-line: no-empty */}
      <HighlightDeleteWrapper onCancel={() => {}} onDelete={() => {}} />
    </TestContainer>);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('properly fire onCancel and onDelete props', () => {
    let deleteClicked = false;
    let cancelClicked = false;

    const component = renderer.create(<TestContainer>
      <HighlightDeleteWrapper
        onCancel={() => { cancelClicked = true; }}
        onDelete={() => { deleteClicked = true; }}
      />
    </TestContainer>);

    renderer.act(() => {
      const deleteButton = component.root.findByProps({ 'data-testid': 'delete' });
      deleteButton.props.onClick();
      const cancelButton = component.root.findByProps({ 'data-testid': 'cancel' });
      cancelButton.props.onClick();
    });

    expect(deleteClicked).toEqual(true);
    expect(cancelClicked).toEqual(true);
  });

  it('focus wrapper on render', async() => {
    const focus = jest.fn();
    const createNodeMock = () => ({ focus });

    renderer.create(<TestContainer>
      <HighlightDeleteWrapper
        onCancel={() => null}
        onDelete={() => null}
      />
    </TestContainer>, {createNodeMock});

    // Wait for React.useEffect()
    // tslint:disable-next-line: no-empty
    await renderer.act(async() => {});

    expect(focus).toHaveBeenCalled();
  });
});
