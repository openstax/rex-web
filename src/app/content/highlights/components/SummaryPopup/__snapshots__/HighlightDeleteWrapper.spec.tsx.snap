// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighlightDeleteWrapper match snapshot 1`] = `
<div
  className="HighlightDeleteWrapper__StyledHighlightDeleteWrapper-gn1gyy-0 lnQzGa"
  data-analytics-region="MH delete"
  tabIndex={0}
>
  <span>
    Are you sure you want to delete this highlight?
  </span>
  <div
    className="styles__HighlightEditButtons-sc-1t1yajj-0 JvKbW"
  >
    <button
      className="Button-ayg7nk-0 fFJxez"
      data-analytics-label="delete"
      data-testid="delete"
      onClick={[Function]}
    >
      Delete
    </button>
    <button
      className="Button-ayg7nk-0 liRzrl"
      data-analytics-label="cancel"
      data-testid="cancel"
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</div>
`;
